### ISSUE TITLE
Output missing from history in Windows Terminal

### ISSUE THREAD

**Issue by robeady** [xonsh/xonsh#3498]
In the latest Windows Terminal preview on windows 10, patches of output from xonsh are missing if you scroll up through history.

This does not occur if xonsh is run in the cmd.exe console host, and it does not occur if I run the same commands with the cmd shell inside Windows Terminal.

## xonfig

I've tried xonsh 0.9.15 and 0.9.9, python 3.7 and python 3.8, prompt_toolkit2 and 3, all seem to exhibit the problem.

<details>

```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.9.15               |
| Git SHA          | feeb6315             |
| Commit Date      | Mar 19 10:50:33 2020 |
| Python           | 3.7.7                |
| PLY              | 3.11                 |
| have readline    | False                |
| prompt toolkit   | 3.0.5                |
| shell type       | prompt_toolkit2      |
| pygments         | 2.6.1                |
| on posix         | False                |
| on linux         | False                |
| on darwin        | False                |
| on windows       | True                 |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

</details>

## Expected Behavior

History is complete

## Current Behavior

History has blank gaps in it (number of lines seems to be correct, they're just empty)

## Steps to Reproduce

This seems to occur in various commands, but as an example, change to a large directory e.g. `C:\windows\system32` and run `ls -la`

**Comment by scopatz** [comment]
Thanks for reporting @robeady! Yeah, the terminal history reader was written with the old cmd.exe in mind.  I haven't worked with the new terminal preview yet.  Any help here would be really appreciated.

**Comment by melund** [comment]
I can't seem to reproduce this. @robeady Can you provide a little more information on what you do?

**Comment by robeady** [comment]
there really isn't much to tell... I open windows terminal, start xonsh, run a command with lots of rapid output, and scroll up the window to find lots of it appeared blank in the scrollback.

this also seems to occur in FluentTerminal and in Hyper

**Comment by melund** [comment]
My version of windows terminal:
Windows Terminal
Version: 0.10.781.0

I haven't seen this at all. What is the output of `which ls` ?

**Comment by robeady** [comment]
The output of `which ls` is not relevant, because it happens for other commands too. If I run
```
$ cd C:\Windows\System32\
$ dir .
```
the same thing happens in Windows terminal and FluentTerminal: scrolling up beyond the first screen or two of output, I see massive gaps of blank lines more than a screen in length. Here's a little example (I expanded the window height to capture this) 
![image](https://user-images.githubusercontent.com/9981949/78820813-5a197e00-79d0-11ea-8856-fb6bd0443fbd.png)

Version info:

Windows Terminal
Version: 0.10.781.0

Fluent Terminal 0.7.0.0

**Comment by robeady** [comment]
And from windows terminal with dir

![image](https://user-images.githubusercontent.com/9981949/78821836-0314a880-79d2-11ea-8eb4-f695e4031c37.png)

**Comment by scopatz** [comment]
Can you try setting `$THREAD_SUBPROCS = False` and seeing if that fixes the issue?

**Comment by scopatz** [comment]
Also, here are the docs: https://xon.sh/envvars.html#thread-subprocs

**Comment by melund** [comment]
Ahh. That explains it. I had `$THREAD_SUBPROCS = False` and enabling reproduces the problem for me. It simply misses to catch some output.

**Comment by melund** [comment]
It doesn't happen in conhost. @scopatz could it be a problem with the new windows terminal, or is more likely to be xonsh ?

**Comment by melund** [comment]
Or maybe we should change the default on Windows. Maybe not very nice but maybe the best option for the majority of users.

**Comment by scopatz** [comment]
Yeah, I think it is an issue with the fact that the tool that reads outputs from the console on windows is likely not meaningful anymore. I would be fine with setting the default to False on Windows.

**Comment by robeady** [comment]
I can confirm that setting `$THREAD_SUBPROCS = False` fixes the issue :)

I tried to read https://xon.sh/envvars.html#thread-subprocs, but I didn't understand any of the trade-offs as written there, so I can't offer an opinion on whether it's sensible to change the default.

**Comment by scopatz** [comment]
Ok, great! Confirmation is enough. That sucks about the docs though :wink:

**Comment by melund** [comment]
Hmm. I really forgot that all [captured subproc](https://xon.sh/tutorial.html#captured-subprocess-with-and) [stops working](https://github.com/xonsh/xonsh/pull/3516/checks?check_run_id=578041884#step:4:128)  if `THREAD_SUBPROC=False`. 

 That is too high a price to pay. @scopatz Would there be a way to this only affected uncaptured sub-procs?

**Comment by scopatz** [comment]
That's basically what threadable predictors do.  They allow you to turn this on/off for certain commands and arguments
